src/BaseHostPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @var bool
+     */
+    private $replace;
+
+    /**
+     * @param UriInterface $baseUri
```

Comment:
sorry, lets forget about the string. it would mean we need the uri factory, or parse the host ourselves. not a good idea.